### PR TITLE
3.10: migrate LDAP/RequestHeader/OpenID CA during upgrade

### PR DESCRIPTION
**Pull request by vrutkovs**
3.9 to 3.10 upgrade doesn't set `ca` field for identity providers, as it is now required in 3.10.

This PR ensures CA is copied on master and `ca` field is set (unless `insecure` is specified) for three providers.

This doesn't need to be present in 3.11, as its a one time action during upgrade.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1639589

### CODE REVIEW

File: roles/lib_utils/filter_plugins/oo_filters.py [hunk 1]
```diff
@@ -708,6 +708,36 @@ def lib_utils_mutate_htpass_provider(idps):
     return idps
 
 
+def lib_utils_mutate_providers_with_ca(idps):
+    '''Updates identityProviders list to mutate CA path'''
+    providers = [
```
Comment (michaelgugino) [inline roles/lib_utils/filter_plugins/oo_filters.py]: nit: immutable list should be tuple.

**Comment by sdodson** [comment]
Did this backport something from release-3.11? We've also got https://bugzilla.redhat.com/show_bug.cgi?id=1641083

**Comment by vrutkovs** [comment]
This is not required for 3.11 as LDAP CA file is now required in 3.10

**Pull request merged**